File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> mp;
        
        int n=nums.size();
        for(int i=0;i<n;i++){
            int num=nums[i];
            int more=target-num;
            //because map return the index if element is find else return the index.
            if(mp.find(more)!=mp.end()){
                return {mp[more],i};
            }
            mp[num]=i;
        }
        return {-1,-1};
    }
};

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    // for i:=0;i<len(nums)-1;i++{
    //     for j:=i+1;j<len(nums);j++{
    //         if nums[i]+nums[j]==target{
    //             return []int{i,j};
    //         }
    //     }
    // }
    // return nil;
    mp:=make(map[int]int);
    for i,num:=range nums{
        mp[num]=i;
    }
    for i,num:=range nums{
        if j,ok:=mp[target-num];ok && i!=j{
            return []int{i,j}
        }
    }
    return nil;
}